invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -409,11 +409,11 @@ def get_subjects(self, obj):
         serialized_subjects = []
         ids = []
         for subject in subjects:
-            sub_text = subject.get("subject")
-            if sub_text:
-                serialized_subjects.append({"subject": sub_text})
+            _id = subject.get("id")
+            if _id:
+                ids.append(_id)
             else:
-                ids.append(subject.get("id"))
+                serialized_subjects.append({"subject": subject.get("subject")})
```

Comment:
Will this access always evaluate to something to be used or should there be a check/default value?

Reply:
The subjects list can be missing (which is handled on line 406). Bus as long as subjects is present, the subject inside is mandatory (it's either we add subjects or we don't)